common/src/dodeca.rs

```diff
@@ -171,234 +174,300 @@ impl Vertex {
-                    result[b][c][a] = v;
-                    result[c][a][b] = v;
-                    result[c][b][a] = v;
                 }
             }
-        }
-        assert_eq!(vertex.next(), None);
-        result
-    };
+            assert_eq!(vertex.next(), None);
+            result
+        })
+    }
 
     /// Whether the determinant of the cube-to-node transform is negative
-    static ref CHUNK_TO_NODE_PARITY: [bool; VERTEX_COUNT] = {
-        let mut result = [false; VERTEX_COUNT];
+    pub fn chunk_to_node_parity() -> &'static [bool; VERTEX_COUNT] {
+        static LOCK: OnceLock<[bool; VERTEX_COUNT]> = OnceLock::new();
+        LOCK.get_or_init(|| {
+            let mut result = [false; VERTEX_COUNT];
 
-        for v in Vertex::iter() {
-            result[v as usize] = math::parity(&v.chunk_to_node_f64());
-        }
+            for v in Vertex::iter() {
+                result[v as usize] = math::parity(&v.chunk_to_node_f64());
+            }
+
+            result
+        })
+    }
+
+    pub fn side_normals_f32() -> &'static [na::Vector4<f32>; SIDE_COUNT] {
+        static LOCK: OnceLock<[na::Vector4<f32>; SIDE_COUNT]> = OnceLock::new();
+        LOCK.get_or_init(|| side_normals_f64().map(|n| n.cast()))
+    }
+
+    pub fn reflections_f32() -> &'static [na::Matrix4<f32>; SIDE_COUNT] {
+        static LOCK: OnceLock<[na::Matrix4<f32>; SIDE_COUNT]> = OnceLock::new();
+        LOCK.get_or_init(|| reflections_f64().map(|n| n.cast()))
+    }
+
+    pub fn dual_to_node_f32() -> &'static [na::Matrix4<f32>; VERTEX_COUNT] {
+        static LOCK: OnceLock<[na::Matrix4<f32>; VERTEX_COUNT]> = OnceLock::new();
+        LOCK.get_or_init(|| dual_to_node_f64().map(|n| n.cast()))
+    }
 
-        result
-    };
+    pub fn node_to_dual_f32() -> &'static [na::Matrix4<f32>; VERTEX_COUNT] {
+        static LOCK: OnceLock<[na::Matrix4<f32>; VERTEX_COUNT]> = OnceLock::new();
+        LOCK.get_or_init(|| node_to_dual_f64().map(|n| n.cast()))
+    }
 
-    static ref SIDE_NORMALS_F32: [na::Vector4<f32>; SIDE_COUNT] = SIDE_NORMALS_F64.map(|n| n.cast());
-    static ref REFLECTIONS_F32: [na::Matrix4<f32>; SIDE_COUNT] = REFLECTIONS_F64.map(|n| n.cast());
-    static ref DUAL_TO_NODE_F32: [na::Matrix4<f32>; VERTEX_COUNT] = DUAL_TO_NODE_F64.map(|n| n.cast());
-    static ref NODE_TO_DUAL_F32: [na::Matrix4<f32>; VERTEX_COUNT] = NODE_TO_DUAL_F64.map(|n| n.cast());
-    static ref DUAL_TO_CHUNK_FACTOR_F32: f32 = *DUAL_TO_CHUNK_FACTOR_F64 as f32;
-    static ref CHUNK_TO_DUAL_FACTOR_F32: f32 = *CHUNK_TO_DUAL_FACTOR_F64 as f32;
+    pub fn dual_to_chunk_factor_f32() -> f32 {
+        static LOCK: OnceLock<f32> = OnceLock::new();
+        *LOCK.get_or_init(|| dual_to_chunk_factor_f64() as f32)
+    }
+
+    pub fn chunk_to_dual_factor_f32() -> f32 {
+        static LOCK: OnceLock<f32> = OnceLock::new();
+        *LOCK.get_or_init(|| chunk_to_dual_factor_f64() as f32)
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::math;
```

Comment:
I think I see the confusion. The reason this line isn't needed on the `master` branch is because `dodeca.rs` as a whole uses the `math` module. However, everything using that `math` module has been moved into the `data` module, so the `dodeca` module no longer uses the `math` module directly. Therefore, the tests have to import it directly instead rather than relying on `super::*` to obtain the `math` module.